vm/systemSmartContracts/delegation.go

```diff
@@ -808,3 +1245,120 @@ func (d *delegation) IsContractEnabled() bool {
 func (d *delegation) IsInterfaceNil() bool {
 	return d == nil
 }
+
+func getTransferBackFromVMOutput(vmOutput *vmcommon.VMOutput) *big.Int {
+	transferBack := big.NewInt(0)
+	for _, outAcc := range vmOutput.OutputAccounts {
+		for _, outTransfer := range outAcc.OutputTransfers {
+			transferBack.Add(transferBack, outTransfer.Value)
+		}
+	}
+
+	return transferBack
+}
+
+func moveNodeFromList(sndList []*NodesData, dstList []*NodesData, key []byte) {
+	var toMoveNodeData *NodesData
+	for i, nodeData := range sndList {
+		if bytes.Equal(nodeData.BLSKey, key) {
+			copy(sndList[i:], sndList[i+1:])
+			toMoveNodeData = nodeData
+			break
+		}
+	}
+
+	dstList = append(dstList, toMoveNodeData)
```

Comment:
Move this line: dstList = append(dstList, nodeData) in 1265 and remove toMoveNodeData from the whole method

Reply:
done